src/libponyc/codegen/genexe.c

```diff
@@ -277,6 +278,9 @@ static bool link_exe(compile_t* c, ast_t* program,
 #ifdef PONY_USE_LTO
     "-flto -fuse-linker-plugin "
 #endif
+#if defined(PLATFORM_IS_ALPINE_LINUX) || defined(PLATFORM_IS_FREEBSD)
```

Comment:
Highly disagree with this, as it also breaks cross-compilation. This should be properly detected through a test checking if a symbol in `<execinfo.h>` symbol without any additional libraries, and then with `-lexecinfo` if it can't.